security/s2a/internal/crypter/counter.go

```diff
@@ -0,0 +1,42 @@
+package crypter
+
+import "errors"
+
+const counterLen = 12
+
+// counter is a 96-bit, little-endian counter.
+type counter struct {
+	value   [counterLen]byte
+	invalid bool
```

Comment:
Can we rename this `hasOverflowed`? I don't think there's any way that a sequence number could be invalid unless it overflows.

Reply:
Done